### ISSUE TITLE
0.17.10, not support NodeJS?

### ISSUE THREAD

**Issue by vkiller** [tinode/tinode-js#52]
**Describe the bug**

Code like below
```
Tinode.setNetworkProviders(require('ws'), require('xmlhttprequest'));
Tinode.setDatabaseProvider(require('fake-indexeddb'));

const tinode = new Tinode(config, () => {
});
tinode.enableLogging(true);
tinode.onDisconnect = (err) => {
  console.log('Disconnect')
};
```

**Go error**
```
Error: Unknown or invalid network transport. Running under Node? Call 'Tinode.setNetworkProviders()'.
    at new u (/Users/tam/Downloads/tok/node_modules/tinode-sdk/umd/tinode.prod.js:1:8302)
    at new v (/Users/tam/Downloads/tok/node_modules/tinode-sdk/umd/tinode.prod.js:1:37152)
    at Object.<anonymous> (/Users/tam/Downloads/tok/examples/test.js:10:16)
    at Module._compile (internal/modules/cjs/loader.js:1072:14)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1101:10)
    at Module.load (internal/modules/cjs/loader.js:937:32)
    at Function.Module._load (internal/modules/cjs/loader.js:778:12)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:76:12)
    at internal/main/run_main_module.js:17:47
```

**Comment by or-else** [comment]
Can you reproduce the problem using `tinode.dev.js` and show the crash log?

**Comment by or-else** [comment]
Do you have 'ws' and 'xmlhttprequest' installed?

**Comment by vkiller** [comment]
> Do you have 'ws' and 'xmlhttprequest' installed?

Yes,I do installed the both

**Comment by vkiller** [comment]
> Can you reproduce the problem using `tinode.dev.js` and show the crash log?

I replace tinode.prod.js content with tinode.dev.js,and got below console
```
/Users/tam/Downloads/tok/node_modules/tinode-sdk/umd/tinode.prod.js:1143
        throw new Error("Unknown or invalid network transport. Running under Node? Call 'Tinode.setNetworkProviders()'.");
        ^

Error: Unknown or invalid network transport. Running under Node? Call 'Tinode.setNetworkProviders()'.
    at new Connection (/Users/tam/Downloads/tok/node_modules/tinode-sdk/umd/tinode.prod.js:1143:15)
    at new Tinode (/Users/tam/Downloads/tok/node_modules/tinode-sdk/umd/tinode.prod.js:4672:26)
    at Object.<anonymous> (/Users/tam/Downloads/tok/examples/test.js:10:16)
    at Module._compile (internal/modules/cjs/loader.js:1072:14)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1101:10)
    at Module.load (internal/modules/cjs/loader.js:937:32)
    at Function.Module._load (internal/modules/cjs/loader.js:778:12)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:76:12)
    at internal/main/run_main_module.js:17:47
Tams-Dusto-Pro:examples tam$ 

```

line 1143
<img width="926" alt="WX20211102-093421@2x" src="https://user-images.githubusercontent.com/15703618/139772477-e5a9a2ae-a19d-40fd-9334-549755dfddbf.png">

line 4672
<img width="791" alt="WX20211102-093526@2x" src="https://user-images.githubusercontent.com/15703618/139772556-8c1d9d0a-9bab-4e8d-8ac8-4444e303361a.png">

**Comment by vkiller** [comment]
and my config set
```
{host:'sandbox.tinode.co',secure:true,appName:'APP_NAME',apiKey:'const API_KEY = 'AQEAAAABAAD_rAp4DJh05a1HAwFT3A6K';
',transport:'wss'}
```

**Comment by or-else** [comment]
> transport:'wss'

`transport:'ws'`

**Comment by vkiller** [comment]
> > transport:'wss'
> 
> `transport:'ws'`

Then I got below error
<img width="1435" alt="WX20211103-081600@2x" src="https://user-images.githubusercontent.com/15703618/139968800-042d30b9-8b37-4084-9b2f-97e614ac2575.png">

**Comment by or-else** [comment]
I personally do not use Tinode in NodeJS environment. Please try debugging it yourself. See what is `WebSocketProvider` and what it expects if `new WebSocketProvider(url)` does not work. 
Add something like `console.log(typeof WebSocketProvider)` right before the `new ...`.

**Comment by vkiller** [comment]
> I personally do not use Tinode in NodeJS environment. Please try debugging it yourself. See what is `WebSocketProvider` and what it expects if `new WebSocketProvider(url)` does not work. Add something like `console.log(typeof WebSocketProvider)` right before the `new ...`.

I debug and found,

`typeof WebSocketProvider` is `undefined`

and seems here also undefined
<img width="794" alt="WX20211103-162608@2x" src="https://user-images.githubusercontent.com/15703618/140029050-6ac72546-aaeb-4cf4-a736-8a457d8fb071.png">

and another part of code seems have no effect 
```
    Tinode.setNetworkProviders = function(wsProvider, xhrProvider) {
      WebSocketProvider = wsProvider;
      XHRProvider = xhrProvider;
    };
```
and there are two definition for `WebSocketProvider`and `XHRProvider`

<img width="479" alt="WX20211103-163135@2x" src="https://user-images.githubusercontent.com/15703618/140029491-b38861a1-8b1f-4862-97fb-357ce976394a.png">
<img width="450" alt="WX20211103-163204@2x" src="https://user-images.githubusercontent.com/15703618/140029513-84745235-1172-4ad0-bfa8-e869e707f051.png">

I think that's a bug,but I am confused,and can't make a PR,can you clarify it?

**Comment by or-else** [comment]
Please check if this fixes it: https://github.com/tinode/tinode-js/commit/0a4ec64562db62267725357a6be1756fbf4018a0

**Comment by vkiller** [comment]
> Please check if this fixes it: [0a4ec64](https://github.com/tinode/tinode-js/commit/0a4ec64562db62267725357a6be1756fbf4018a0)

Yes,that works. the npm should be updated,thanks.

**Comment by or-else** [comment]
Released in v0.18.0-beta1
